components/director/internal/domain/application/repository_test.go

```diff
@@ -15,39 +20,319 @@ import (
+		}}
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			//GIVEN
+			sqlxDB, sqlMock := testdb.MockDatabase(t)
+			defer sqlMock.AssertExpectations(t)
+			ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+
+			pgRepository := application.NewPostgresRepository(conv)
+
+			expectedQuery := fmt.Sprintf(pageableQuery, testCase.ExpectedLimit, testCase.ExpectedOffset)
+			sqlMock.ExpectQuery(expectedQuery).
+				WithArgs(tenantID).
+				WillReturnRows(testCase.Rows)
+
+			countRow := sqlMock.NewRows([]string{"count"}).AddRow(testCase.TotalCount)
+			sqlMock.ExpectQuery(countQuery).
+				WithArgs(tenantID).
+				WillReturnRows(countRow)
+
+			//THEN
+			modelPage, err := pgRepository.List(ctx, tenantID, nil, testCase.InputPageSize, testCase.InputCursor)
+
+			//THEN
+			require.NoError(t, err)
+			assert.Equal(t, testCase.ExpectedLimit, modelPage.TotalCount)
+			require.NoError(t, sqlMock.ExpectationsWereMet())
+
+			assert.Equal(t, app1ID, modelPage.Data[0].ID)
+			assert.Equal(t, tenantID, modelPage.Data[0].Tenant)
+
+			assert.Equal(t, app2ID, modelPage.Data[1].ID)
+			assert.Equal(t, tenantID, modelPage.Data[1].Tenant)
+
+		})
+	}
+
+	t.Run("Returns error when decoded cursor is non-positive number", func(t *testing.T) {
+		//GIVEN
+		sqlxDB, sqlMock := testdb.MockDatabase(t)
+		defer sqlMock.AssertExpectations(t)
+		ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+		pgRepository := application.NewPostgresRepository(conv)
+		//THEN
+		_, err := pgRepository.List(ctx, tenantID, nil, 2, pagination.EncodeNextOffsetCursor(1, -3))
+
+		//THEN
+		require.EqualError(t, err, "while decoding page cursor: cursor is not correct")
+	})
+}
+
+func TestPgRepository_GetByID_ShouldReturnApplicationModelForApplicationEntity(t *testing.T) {
+	// given
+	conv := application.NewConverter(nil, nil, nil, nil)
+	tenantID := uuid.New().String()
+	appID := uuid.New().String()
+
+	timestamp, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	require.NoError(t, err)
+
+	sqlxDB, sqlMock := testdb.MockDatabase(t)
+	defer sqlMock.AssertExpectations(t)
+
+	rows := sqlmock.NewRows([]string{"id", "tenant_id", "name", "description", "status_condition", "status_timestamp", "healthcheck_url"}).
+		AddRow(appID, tenantID, "App ABC", "Description for application ABC", "INITIAL", timestamp, "http://domain.local")
+
+	sqlMock.ExpectQuery(`^SELECT (.+) FROM public.applications WHERE tenant_id = \$1 AND id = \$2$`).
+		WithArgs(tenantID, appID).
+		WillReturnRows(rows)
+
+	ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+
+	pgRepository := application.NewPostgresRepository(conv)
+
+	// when
+	modelApp, err := pgRepository.GetByID(ctx, tenantID, appID)
+
+	//then
+	require.NoError(t, err)
+	assert.NoError(t, sqlMock.ExpectationsWereMet())
+	assert.Equal(t, appID, modelApp.ID)
+	assert.Equal(t, tenantID, modelApp.Tenant)
+}
+
+func TestPgRepository_Create_ShouldCreateApplicationEntityFromValidModel(t *testing.T) {
```

Comment:
Test the following cases:
- Success
- Converter Error
- DB Error

Please see document / fetch request repository tests.